extension/js/content_scripts/webmail/gmail-element-replacer.ts

```diff
@@ -473,21 +482,27 @@ export class GmailElementReplacer implements WebmailElementReplacer {
         messageInfo,
         skipGoogleDrive
       );
-      if (renderStatus === 'hidden') {
+      if (this.isAttachmentHideable(a, renderStatus)) {
         nRenderedAttachments--;
       }
     }
     if (nRenderedAttachments !== attachments.length) {
       // according to #4200, no point in showing "download all" button if at least one attachment is encrypted etc.
       $(this.sel.attachmentsButtons).hide();
     }
-    if (nRenderedAttachments >= 2) {
-      // Aligned with Gmail, the label is shown only if there are 2 or more attachments
-      attachmentsContainerInner.parent().find(this.sel.numberOfAttachmentsDigit).text(nRenderedAttachments);
-      attachmentsContainerInner.parent().find(this.sel.numberOfAttachments).show();
-    }
     if (nRenderedAttachments === 0) {
       attachmentsContainerInner.parents(this.sel.attachmentsContainerOuter).first().hide();
+    } else {
+      const elementsToClone = ['span.a2H', 'div.a2b'];
+      const scannedByGmailLabel = $(elementsToClone[0]).first().clone();
+      const scannedByGmailPopover = $(elementsToClone[1]).first().clone(true);
```

Comment:
maybe we can just update text in existing html elements, instead of cloning them and then re-appending?

Reply:
>  I just noticed a small bug 

Thank you - I'll look into it in and will fix that as well.

> maybe we can just update text in existing html elements, instead of cloning them and then re-appending?

Actually, that was my first attempt in fixing the issue but the HTML structuring in Gmail is not scraping-friendly, I mean, the text node and html elements kinda mixed up in a single span so simple text content changing for example via `.text()` don't work very great. It's a bit complex for example, the word/label "One attachment" appears for unmodified number of attachments, wherein with modified number of attachments - they will be defined in a numerical representation which conflicts with the HTML structuring so I came up with element cloning so I can have the better HTML structure I wanted to have + the uniformity of the number of attachments is pleasant compared to modifying "One" to "1" is again complex because the word "One" is a text node and "1" however, is nicely wrapped in an element. 

The live gmail test is not passing at all. Was the test in the web extension page maybe sufficient enough as I could add alternative for testing the expected numbers?

Reply:
> Actually, that was my first attempt in fixing the issue but the HTML structuring in Gmail is not scraping-friendly, I mean, the text node and html elements kinda mixed up in a single span so simple text content changing for example via .text() don't work very great. It's a bit complex for example, the word/label "One attachment" appears for unmodified number of attachments, wherein with modified number of attachments - they will be defined in a numerical representation which conflicts with the HTML structuring so I came up with element cloning so I can have the better HTML structure I wanted to have + the uniformity of the number of attachments is pleasant compared to modifying "One" to "1" is again complex because the word "One" is a text node and "1" however, is nicely wrapped in an element.

Got it, thanks for detailed explanation, then let's use your current implementation 👍

> The live gmail test is not passing at all. Was the test in the web extension page maybe sufficient enough as I could add alternative for testing the expected numbers?

Having live gmail test will be useful for detecting if there will be any changes in Google's html markup, which can lead to breaking of our html parsing. We can use existing tests, instead of creating a separate one:
- `mail.google.com - attachments which contain emoji in filename are rendered correctly` - here let's check that page has label `2 Attachments  •  Scanned by Gmail`
- `mail.google.com - pubkey file gets rendered` - here check that attachments count label is hidden

Reply:
> Works much better now, well done, I just noticed a small bug

Thank you @sosnovsky. It is now fixed.

 I have made changes that would satisfy all this. Test are all passing now.

https://flowcrypt.semaphoreci.com/workflows/5ebecf93-c789-40ad-888d-ce64a720642e?pipeline_id=2fa93122-d051-44d3-855d-94cd5319d44b